src/textual/widgets/_input.py

```diff
@@ -332,7 +332,7 @@ async def _on_key(self, event: events.Key) -> None:
             event.prevent_default()
 
     def _on_paste(self, event: events.Paste) -> None:
-        line = event.text.splitlines()[0]
+        line = event.text.splitlines()[0] if event.text else ""
```

Comment:
Could we not skip the `insert_text_at_cursor` entirely when the pasted text is an empty string.

I'm assuming that inserting an empty string does nothing useful.

Reply:
It doesn't do anything very useful, but that would short-circuit other things that might be expected, like an `Input.Changed` event.
Do you think it's worth adding that exception here?

Reply:
I don't think you would get a `Change` event, because nothing should have changed. Have you checked?

Reply:
My bad, I thought `always_update` was `True` for reactives by default.